spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JdbcEnvironmentRepository.java

```diff
@@ -119,15 +119,21 @@ public Environment findOne(String application, String profile, String label) {
 		List<String> envs = new ArrayList<>(new LinkedHashSet<>(Arrays.asList(profiles)));
 		Collections.reverse(applications);
 		Collections.reverse(envs);
-		for (String env : envs) {
-			for (String app : applications) {
-				addPropertySource(environment, app, env, label);
-			}
+		String[] labels = { label };
+		if (labels[0].contains(",")) {
+			labels = labels[0].split(",");
```

Comment:
Maybe just like `applications` (line 120) and `profiles` (line 121), `labels` also should be reversed?  for given label string of `L1,L2,L3`, `L3` should be the first one to query, followed by `L2` and finally `L1`.

Reply:
Good point I have made that change